durable-storage/src/merkle_layer/node.rs

```diff
@@ -0,0 +1,486 @@
+// SPDX-FileCopyrightText: 2025 Trilitech <contact@trili.tech>
+//
+// SPDX-License-Identifier: MIT
+
+use std::cmp::Ordering;
+use std::fmt::Debug;
+use std::sync::Arc;
+
+use super::Key;
+
+/// A node that supports rebalancing and Merklisation.
+#[derive(Clone, Default, Debug)]
+pub(super) struct MavlNode {
+    key: Key,
+    data: Vec<u8>,
```

Comment:
You clone this feel very often. I would advocate using something that is more lenient regarding the cost of cloning. [`Bytes`](https://docs.rs/bytes/latest/bytes/struct.Bytes.html) could be a good option.